src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakService.kt

```diff
@@ -65,30 +65,60 @@ class VedtakService(private val vedtakRepository: VedtakRepository) {
         return null
     }
 
-    fun hentForventetInntektForBehandlingIds(behandlingIds: Collection<UUID>): Map<UUID, Int?> {
+    fun hentForventetInntektForBehandlingIds(behandlingIds: Collection<UUID>): List<ForventetInntektForBehandling> {
         val vedtakList = vedtakRepository.findAllById(behandlingIds)
-        val dagensDatoMinusEnMåned = LocalDate.now().minusMonths(1)
-        val map = mutableMapOf<UUID, Int?>()
+        val list = mutableListOf<ForventetInntektForBehandling>()
         for (vedtak in vedtakList) {
             if (vedtak.erVedtakAktivtForDato(LocalDate.now())) {
-                map.put(vedtak.behandlingId, vedtak.inntekter?.inntekter?.firstOrNull {
-                    dagensDatoMinusEnMåned.isEqualOrAfter(it.startDato) && dagensDatoMinusEnMåned.isEqualOrBefore(it.sluttDato)
-                }?.inntekt?.toInt())
+                list.add(createForventetInntektForBehandling(vedtak))
             } else {
-                map.put(vedtak.behandlingId, null)
+                list.add(ForventetInntektForBehandling(vedtak.behandlingId, null, null))
             }
         }
 
-        return map
+        return list
     }
 
+    private fun createForventetInntektForBehandling(vedtak: Vedtak): ForventetInntektForBehandling {
+        return ForventetInntektForBehandling(vedtak.behandlingId,
+                                             createForventetInntektForMåned(vedtak, LocalDate.now().minusMonths(1)),
+                                             createForventetInntektForMåned(vedtak, LocalDate.now().minusMonths(2))
+        )
+    }
+
+    private fun createForventetInntektForMåned(vedtak: Vedtak, forventetInntektForDato: LocalDate) =
+            vedtak.inntekter?.inntekter?.firstOrNull {
```

Comment:
Personer som fått en revurdering frem i tiden kommer ikke bli oppdaget at de har 10% økning før man kommit til den måneden de er revurdert fra. Er kanskje ingen stor gruppe
Burde kunnet hentet tilkjent ytelser her og, då den har andeler som inneholder inntekten for hvert måned

Reply:
Angående revurderinger frem i tid: Det gjøres en sjekk på om vedtak er aktivt, dersom det ikke er aktivt returneres null og det tas ikke med i beregningen av i personhendelse.
Har gjort om koden til å ta i bruk tilkjent ytelser.